node/primitives/src/lib.rs

```diff
@@ -42,54 +37,23 @@ pub enum Statement {
 	Invalid(Hash),
 }
 
-impl Statement {
-	/// Get the signing payload of the statement.
-	pub fn signing_payload(&self, context: &SigningContext) -> Vec<u8> {
-		// convert to fully hash-based payload.
+impl EncodeAs<CompactStatement> for Statement {
+	fn encode_as(&self) -> Vec<u8> {
 		let statement = match *self {
-			Statement::Seconded(ref c) => polkadot_primitives::parachain::Statement::Candidate(c.hash()),
-			Statement::Valid(hash) => polkadot_primitives::parachain::Statement::Valid(hash),
-			Statement::Invalid(hash) => polkadot_primitives::parachain::Statement::Invalid(hash),
+			Statement::Seconded(ref c) => {
+				polkadot_primitives::parachain::CompactStatement::Candidate(c.hash())
+			}
+			Statement::Valid(hash) => polkadot_primitives::parachain::CompactStatement::Valid(hash),
+			Statement::Invalid(hash) => polkadot_primitives::parachain::CompactStatement::Invalid(hash),
 		};
-
-		statement.signing_payload(context)
+		statement.encode()
 	}
 }
 
 /// A statement, the corresponding signature, and the index of the sender.
 ///
 /// Signing context and validator set should be apparent from context.
-#[derive(Debug, Clone, PartialEq, Encode, Decode)]
-pub struct SignedStatement {
-	/// The statement signed.
-	pub statement: Statement,
-	/// The signature of the validator.
-	pub signature: ValidatorSignature,
-	/// The index in the validator set of the signing validator. Which validator set should
-	/// be apparent from context.
-	pub sender: ValidatorIndex,
-}
-
-impl SignedStatement {
-	/// Check the signature on a statement. Provide a list of validators to index into
-	/// and the context in which the statement is presumably signed.
-	///
-	/// Returns an error if out of bounds or the signature is invalid. Otherwise, returns Ok.
-	pub fn check_signature(
-		&self,
-		validators: &[ValidatorId],
-		signing_context: &SigningContext,
-	) -> Result<(), ()> {
-		let validator = validators.get(self.sender as usize).ok_or(())?;
-		let payload = self.statement.signing_payload(signing_context);
-
-		if self.signature.verify(&payload[..], validator) {
-			Ok(())
-		} else {
-			Err(())
-		}
-	}
-}
+pub type SignedStatement = Signed<Statement, CompactStatement>;
```

Comment:
We need to be careful that runtime APIs accept `Signed<CompactStatement>`, not `Signed<Statement>`. 

I propose calling this `SignedFullStatement`